### PR TITLE
Check if error during test after opening a file

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: #1074

### CODE REVIEW

**Comment by ahsonkhan** [comment]
Need to fix this instance as well, which is where the original issue came from:
https://github.com/Azure/azure-sdk-for-cpp/blob/674a8b623a4ba86172b481af6e63a59cac1612a2/sdk/core/azure-core/test/e2e/azure_core_with_curl_bodyBuffer.cpp#L83-L84

https://github.com/Azure/azure-sdk-for-cpp/pull/1070#discussion_r535383111

**Review by ahsonkhan — APPROVED** [review]
Other than making a similar fix in `azure_core_with_curl_bodyBuffer.cpp`, looks good.

File: sdk/core/azure-core/test/e2e/azure_core_with_curl_bodyBuffer.cpp [hunk 1]
```diff
@@ -72,18 +72,41 @@ int main()
...
 void doFileRequest(Context const& context, HttpPipeline& pipeline)
 {
 
   Azure::Core::Http::Url host("https://httpbin.org/put");
   cout << "Creating a Put From File request to" << endl
        << "Host: " << host.GetAbsoluteUrl() << endl;
 
+#if defined(AZ_PLATFORM_POSIX)
   // Open a file that contains: {{"key":"value"}, {"key2":"value2"}, {"key3":"value3"}}
   int fd = open("/home/vivazqu/workspace/a", O_RDONLY);
```
Comment (ahsonkhan) [inline sdk/core/azure-core/test/e2e/azure_core_with_curl_bodyBuffer.cpp:84]: Btw, hardcoding a local file path specific to your machine isn't great. We should fix this.

File: sdk/core/azure-core/test/e2e/azure_core_with_curl_bodyBuffer.cpp [hunk 2]
```diff
@@ -72,18 +72,41 @@ int main()
...
+  if (fd < 0)
+  {
+    throw std::invalid_argument("Fail to open the requested file");
+  }
   // Create Stream from file starting with offset 18 to 100
   auto requestBodyStream = FileBodyStream(fd, 18, 100);
+#elif defined(AZ_PLATFORM_WINDOWS)
+  // NOTE: To run the sample: Create folder 'home' on main hard drive (like C:/) and then add a file
+  // `a` in there
+  //
```
Comment (ahsonkhan) [inline sdk/core/azure-core/test/e2e/azure_core_with_curl_bodyBuffer.cpp:94]: Running tests and samples shouldn't require manual artifact creation on the machine. Especially not in directories external to the repo.

The test should create whatever file or directory it needs within the sub-folder inside the repo and use it.

**Review by ahsonkhan — CHANGES_REQUESTED** [review]
Fix the e2e sample/test so that it is runnable on anyone's machine or remove it if not needed or useful.

Alternatively, leave it untouched in this PR, if you feel it is out of scope (and only fix the adapter test), and address it separately. Up to you.

**Comment by vhvb1989** [comment]
> Fix the e2e sample/test so that it is runnable on anyone's machine or remove it if not needed or useful.
> 
> Alternatively, leave it untouched in this PR, if you feel it is out of scope (and only fix the adapter test), and address it separately. Up to you.

removing e2e. https://github.com/Azure/azure-sdk-for-cpp/issues/1214

**Review by ahsonkhan — APPROVED** [review]

**Pull request merged**